newIDE/app/src/MainFrame/RouterContext.js

```diff
@@ -0,0 +1,83 @@
+// @flow
+import * as React from 'react';
+import Window, { type AppArguments } from '../Utils/Window';
+
+export type Router = {|
+  routeArguments: AppArguments,
+  removeRouteArguments: (string[]) => void,
+  addRouteArguments: AppArguments => void,
+  navigateToRoute: (route: string, additionalArgument?: AppArguments) => void,
+|};
+
+const initialState: Router = {
+  routeArguments: {},
+  removeRouteArguments: () => {},
+  addRouteArguments: () => {},
+  navigateToRoute: () => {},
+};
+
+const RouterContext = React.createContext<Router>(initialState);
+
+export default RouterContext;
+
+type Props = {|
+  children?: React.Node,
+|};
+
+export const RouterContextProvider = ({ children }: Props) => {
+  const initialWindowArguments = Window.getArguments();
+  // Put value in the state, so we can control when the DOM re-renders.
+  const [routeArguments, setRouteArguments] = React.useState(
+    initialWindowArguments
+  );
+
+  const removeRouteArguments = React.useCallback(
+    (argumentsToRemove: string[]) => {
+      // Remove them from the window. (only for web)
+      Window.removeArguments(argumentsToRemove);
+      // Update the state accordingly, based on the previous state.
+      setRouteArguments(oldArguments => {
+        const newArguments = { ...oldArguments };
+        argumentsToRemove.forEach(argument => {
+          delete newArguments[argument];
+        });
+        return newArguments;
+      });
+    },
+    []
+  );
+
+  const addRouteArguments = React.useCallback(
+    (argumentsToAdd: AppArguments) => {
+      // Add them to the window. (only for web)
+      Window.addArguments(argumentsToAdd);
+      // Update the state accordingly, based on the previous state.
+      setRouteArguments(oldArguments => ({
+        ...oldArguments,
+        ...argumentsToAdd,
+      }));
+    },
+    []
+  );
+
+  const navigateToRoute = React.useCallback(
+    (route: string, additionalArguments?: AppArguments) => {
```

Comment:
It feels weird to have a free argument here.
In server side rendering, you generally have a Route object and you can call a method `toPath` on it so that there is no typo issue (and maybe handle other issues). It's the same issue as for the `ROUTING_ARGUMENTS` I think.
But it could be an improvement for later